src/gsd.js

```diff
@@ -125,40 +137,51 @@ function gsd(x, yIn, options) {
     intervalR.length = intervalRLen;
     broadMask.length = broadMaskLen;
 
-    var signals = new Array(minddY.length);
-    var signalsLen = 0;
-    var lastK = 0;
-    var possible, frequency, distanceJ, minDistance, gettingCloser;
+    let signals = new Array(minddY.length);
+    let signalsLen = 0;
+    let lastK = 0;
+    let possible, frequency, distanceJ, minDistance, gettingCloser;
     for (let j = 0; j < minddY.length; ++j) {
         frequency = X[minddY[j]];
         possible = -1;
         let k = lastK + 1;
         minDistance = Number.MAX_VALUE;
         distanceJ = 0;
         gettingCloser = true;
-        while (possible === -1 && k < intervalL.length && gettingCloser) {
-            distanceJ = Math.abs(frequency - (intervalL[k] + intervalR[k]) / 2);
+        while (possible === -1 && (k < intervalL.length) && gettingCloser) {
+            distanceJ = Math.abs(frequency - (intervalL[k].val + intervalR[k].val) / 2);
+
             //Still getting closer?
             if (distanceJ < minDistance) {
                 minDistance = distanceJ;
             } else {
                 gettingCloser = false;
             }
-            if (distanceJ < Math.abs(intervalL[k] - intervalR[k]) / 2) {
+            if (distanceJ < Math.abs(intervalL[k].val - intervalR[k].val) / 2) {
                 possible = k;
                 lastK = k;
             }
-            k++;
+            ++k;
```

Comment:
In this case there is no difference between ++k and k++. Because in both cases you increase and assign to the same variable. It is only different when you assign to a different variable:
var b=0;
var a = b++;    //a = 0
is different than:
var b=0;
var a = ++b; //a = 1